### DISCUSSION TITLE
An Introduction to Apple Pkl: A New Open-Source Configuration Language

### DISCUSSION THREAD

**Discussion by narmidm** [apple/pkl#133] [Show and tell]
Hi everyone,

I wrote an article about Apple Pkl, a new open source configuration language developed by Apple. In this article, I explain what Pkl is, why it is useful, and how it compares to other configuration languages like JSON, YAML, or Property Lists. I also provide some examples of Pkl code and its benefits.

You can read the article here: <a href="https://medium.com/wisemonks/apple-pkl-a-new-programming-language-for-configuration-6a8d1d2c903e" target="_blank">Apple Pkl: A New Programming Language for Configuration</a>

I would love to hear your feedback and thoughts on Pkl and my article.

**Comment by john-aws** [comment]
Thanks. Any idea why it is named `Pkl`?

**Reply by narmidm** [reply]
PKL is pronounced as Pickle. I have heard that PKL means _**Python-Kotlin-Lisp**_, and I have also heard that it means **_Programming Konfiguration Language_**. However, there is nothing in the official documentation about it. Maybe they will tell us in the future. If anyone knows, please let us know. I am also eager to find out 😄

**Comment by bioball** [comment]
Thanks for sharing!

**Comment by opentokix** [comment]
https://xkcd.com/927/
